types/node/index.d.ts

```diff
@@ -38,6 +38,7 @@
 //                 Jordi Oliveras Rovira <https://github.com/j-oliveras>
 //                 Thanik Bhongbhibhat <https://github.com/bhongy>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
```

Comment:
The build now errors with:

> Error: accepts depends on node but has a lower required TypeScript version.

What to do?

The reason I had to add this was because without it, `npm run lint node` resulted in lots of weird errors. 🤷‍♂️ E.g.:

```
Error: /Users/oliverjash/Development/DefinitelyTyped/types/node/node-tests.ts:510:15
ERROR: 510:15  expect  Compile error in typescript@2.0 but not in typescript@2.1.
Fix with a comment '// TypeScript Version: 2.1' just under the header.
Type 'string' is not assignable to type 'UrlObject[]'.
ERROR: 510:15  expect  Compile error in typescript@2.0 but not in typescript@2.1.
Fix with a comment '// TypeScript Version: 2.1' just under the header.
Cannot redeclare block-scoped variable 'urlObjectQuery'.
…
```

These seem like invalid errors, and I'm not sure how to workaround this.

Reply:
We cannot change the min version here as too many packages rely on it.

Reply:
Unless you are willing to update all dependant packages.

Reply:
Can you advise on the 2.0 lint errors?

Reply:
TS 2.1 has introduced keyof and Lookup Types which is needed for e.g. `url.UrlObject['query']`. You could use
```
const urlObjectQuery: string | null | querystring.ParsedUrlQueryInput = {};
// $ExpectError
const urlQuery: string | null | querystring.ParsedUrlQuery = urlObjectQuery;
```

Reply:
Thanks @Flarna. It doesn't test quite the same thing, but it will do!